### ISSUE TITLE
`misc-const-correctness` not detected for `std::vector` with `front()` call

### ISSUE THREAD

**Issue by firewave** [llvm/llvm-project#58246]
```cpp
#include <vector>

struct S {
    std::vector<S> cb();

    void d() {
        std::vector<S> ft1 = cb();
        (void)ft1.front();
    }
};
```
https://godbolt.org/z/qYGE4j44b

CC @JonasToth

**Comment by vbvictor** [comment]
Same underlying problem as in https://github.com/llvm/llvm-project/issues/175033.
I will close in favor that issue.
